### PR TITLE
[FEAT] send clean backtrace if applicable

### PR DESCRIPTION
**Pull request by elyara**
Start using the active_support built-in way to clean the backtrace
tested in **v2** & **houston** & **moderation_chimera**

### CODE REVIEW

File: graylog2_exceptions.gemspec [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 Gem::Specification.new do |s|
   s.name = %q{friendlyfashion-graylog2_exceptions}
-  s.version = "1.3.2"
```
Comment (oded-zahavi) [inline graylog2_exceptions.gemspec:8]: Why not align with gem format, that is have a dedicated file to hold version.

File: graylog2_exceptions.gemspec [hunk 2]
```diff
@@ -26,7 +26,7 @@ Gem::Specification.new do |s|
   ]
   s.homepage = %q{http://www.graylog2.org/}
   s.require_paths = ["lib"]
-  s.rubygems_version = %q{1.3.7}
+  s.rubygems_version = %q{1.8.3}
```
Comment (oded-zahavi) [inline graylog2_exceptions.gemspec]: Is it? Ruby gems version?

File: lib/graylog2_exceptions.rb [hunk 3]
```diff
@@ -165,7 +165,25 @@ def clean_stack(backtrace)
...
+  def clean_stack(backtrace)
+    backtrace_cleaner.clean(backtrace).join("\n")
+  end
+
+  def backtrace_cleaner
+    @backtrace_cleaner ||=get_backtrace_cleaner
+    @backtrace_cleaner
+  end
+
+  def get_backtrace_cleaner
```
Comment (oded-zahavi) [inline lib/graylog2_exceptions.rb]: I'd implement it in backtrace_cleaner. There's no real reason to do it in a different method

File: lib/backtrace_cleaner.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+module Fiverr
+  class BacktraceCleaner < ActiveSupport::BacktraceCleaner
+    APP_DIRS_PATTERN        = /^\/?(app|config|lib|test)/
+    FIVERR_GEMS_PATTERN     = /.*bundler\/gems/
+    RENDER_TEMPLATE_PATTERN = /:in `_render_template_\w*'/
+
+    def initialize
+      super
+      add_filter   { |line| line.sub("#{Rails.root}/", '') }
```
Comment (oded-zahavi) [inline lib/backtrace_cleaner.rb]: I assume add_filter is implemented in base class?

File: lib/graylog2_exceptions.rb [hunk 5]
```diff
@@ -165,7 +165,25 @@ def clean_stack(backtrace)
...
+    if defined? ActiveSupport::BacktraceCleaner
+      require_relative './backtrace_cleaner'
+      Fiverr::BacktraceCleaner.new
+    else
+      self
```
Comment (oded-zahavi) [inline lib/graylog2_exceptions.rb:182]: Do Fiverr::BacktraceCleaner and Graylog2Exceptions implement the same interface?

File: lib/graylog2_exceptions.rb [hunk 6]
```diff
@@ -165,7 +165,25 @@ def clean_stack(backtrace)
...
+    arr
+  end
+
+  def clean_stack(backtrace)
+    backtrace_cleaner.clean(backtrace).join("\n")
+  end
+
+  def backtrace_cleaner
+    @backtrace_cleaner ||=get_backtrace_cleaner
+    @backtrace_cleaner
```
Comment (oded-zahavi) [inline lib/graylog2_exceptions.rb]: Redundant

File: lib/backtrace_cleaner.rb (hunk 4 above)
Comment (elyara) [reply lib/backtrace_cleaner.rb]: you assume right

File: lib/graylog2_exceptions.rb (hunk 5 above)
Comment (elyara) [reply lib/graylog2_exceptions.rb:182]: only what i'm interested in: `clean`

**Pull request merged**